### PR TITLE
Bump Mcore to 7f7439f

### PR DESCRIPTION
**Pull request by chtruong814**
# What does this PR do ?

Bump Mcore to 0c6c117

# Changelog

- Bump Mcore to f8d3e2e

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/modelopt/distill/utils.py [hunk 1]
```diff
@@ -279,7 +277,6 @@ def adjust_tensor_shapes(recv_tensor_shapes: List[Tuple[int, ...]], send_tensor_
             micro_batch_size=micro_batch_size,
             decoder_seq_length=decoder_seq_length,
             config=teacher_config,
-            encoder_decoder_xattn=teacher_encoder_decoder_xattn,
```
Comment (chtruong814) [inline nemo/collections/llm/modelopt/distill/utils.py:282]: @AAnoosheh I'm working on updating Mcore on Github NeMo and some things are breaking.  It looks like the kwarg `encoder_decoder_xattn` was removed in a recent Mcore update.  Any concerns with this change?

https://github.com/NVIDIA/Megatron-LM/commit/b600e38d7b2a5b31d8d90e35bcf0cad18977a99c#diff-c27f9f2765a43b5c781756ce9fe4b9abcc3618ec96157f5dc67c4a00a7900b73R177-R1589

File: nemo/collections/vlm/inference/llava_inference_wrapper.py [hunk 2]
```diff
@@ -94,8 +93,9 @@ def forward_pass_without_pipeline_parallel(self, inference_input: Dict[str, Any]
         Returns:
             torch.Tensor: The output logits of shape [batch_size, seq_len, padded_vocab_size]
         """
-        logits = self.model(attention_mask=None, inference_params=self.inference_params, **inference_input)
-        logits = tensor_parallel.gather_from_tensor_model_parallel_region(logits)
+        logits = self.model(
+            attention_mask=None, inference_params=self.inference_params, runtime_gather_output=True, **inference_input
+        )
```
Comment (chtruong814) [inline nemo/collections/vlm/inference/llava_inference_wrapper.py:98]: @yaoyu-33 I'm working on bumping Mcore here, and some things broke.  One of the recent Mcore changes seems to require `runtime_gather_output=True` if it's in inference mode.  Please review to see if this change looks ok to you.

https://github.com/NVIDIA/Megatron-LM/commit/d5ef16eab4693a593a672f162fb6b2197904e773

File: nemo/collections/llm/modelopt/distill/utils.py (hunk 1 above)
Comment (AAnoosheh) [reply nemo/collections/llm/modelopt/distill/utils.py:282]: None – all good. Thanks

File: nemo/collections/vlm/inference/vlm_inference_controller.py [hunk 3]
```diff
@@ -62,12 +62,20 @@ def tokenize_prompt(self, prompt: str, image):
...
+        use_attention_mask: bool = False,
+    ):
         """Preparing input data for inference, using respective wrapper's prep_inference_input method
 
         Args:
             prompts_tokens (torch.Tensor): A tensor of shape [batch_size, max_sequence_length]
             active_requests (OrderedDict[int, InferenceRequest]): The input active requests
+            use_attention_mask (bool): Whether to use an attention mask. Should be set to True only
+                when exclusively doing prefill (no decode) with variable prompt lengths.
+                Currently unused and added to match an expected interface in Mcore.
```
Comment (chtruong814) [inline nemo/collections/vlm/inference/vlm_inference_controller.py:78]: @yaoyu-33 As part of updating Mcore on NeMo, it looks like, this method now is expecting the kwarg `use_attention_mask`.  I made the test "pass" by just adding it, but unsure if we should actually try to do something with it.  

https://github.com/NVIDIA/Megatron-LM/commit/14a025015e9b7c05cdaadc2cc3e3dc85b4c4302b

Please take a look and let me know your thoughts.

**Review by yaoyu-33 — DISMISSED** [review]

**Review by yaoyu-33 — APPROVED** [review]

**Review by pablo-garay — APPROVED** [review]
